viz_scripts/generic_metrics.ipynb

```diff
@@ -101,192 +102,353 @@
+    "        if not df_sensed_dist_land.empty:\n",
+    "            bar_count += 1\n",
+    "            all_data_frames.append(df_sensed_dist_land)\n",
+    "    else:\n",
+    "        print(\"expanded_ct_sensed is empty.\")\n",
+    "\n",
+    "    result_df = merge_dataframes(all_data_frames)\n",
+    "    stacked_bar_chart_generic(plot_title, result_df, file_name, bar_count)\n",
+    "    alt_text, alt_html = store_alt_text_stacked_bar_chart(result_df[result_df['Count'] > 0], file_name, plot_title)\n",
     "except:\n",
-    "    d10_df = expanded_ct.query(\"distance <= \" + str(cutoff)) if \"distance\" in expanded_ct.columns else expanded_ct\n",
-    "    debug_df.loc[\"Trips_less_than_80th_pct\"] = scaffolding.trip_label_count(\"Mode_confirm\", d10_df)\n",
-    "    generate_missing_plot(plot_title_no_quality,debug_df,file_name)\n",
-    "    alt_text = store_alt_text_missing(debug_df, file_name, plot_title_no_quality)"
+    "    generate_missing_plot(plot_title_no_quality, merged_debug_df, file_name)\n",
+    "    alt_text = store_alt_text_missing(merged_debug_df, file_name, plot_title_no_quality)"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "dominant-company",
+   "id": "ffb2df0b",
    "metadata": {},
    "source": [
-    "### Miles per chosen transport mode"
+    "### 5. 100% Stacked Bar Charts (Count by Purpose)"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "satisfied-sharing",
-   "metadata": {
-    "scrolled": false
-   },
+   "id": "46474ada",
+   "metadata": {},
    "outputs": [],
    "source": [
-    "plot_title_no_quality= label_units + \" for each mode (selected by users)\"\n",
-    "plot_title=plot_title_no_quality + '\\n' + quality_text\n",
-    "file_name ='miles_mode_confirm%s' % file_suffix\n",
+    "plot_title_no_quality = \"Number of trips for each purpose (selected by users)\"\n",
+    "file_name = f\"ntrips_purpose{file_suffix}\"\n",
     "\n",
     "try:\n",
-    "    dist = expanded_ct.groupby('Mode_confirm').agg({distance_col: ['sum', 'count' , 'mean']})\n",
-    "    dist.columns = ['Total ('+label_units_lower+')', 'Count', 'Average ('+label_units_lower+')']\n",
-    "    dist = dist.reset_index()\n",
-    "    dist =dist.sort_values(by=['Total ('+label_units_lower+')'], ascending=False)\n",
-    "    dist_dict = dict(zip(dist['Mode_confirm'], dist['Total ('+label_units_lower+')']))\n",
-    "    \n",
-    "    labels_m = []\n",
-    "    values_m = []\n",
-    "    \n",
-    "    for x, y in dist_dict.items():\n",
-    "        labels_m.append(x)\n",
-    "        values_m.append(y)\n",
+    "    bar_count = 0\n",
+    "    all_tp_data_frames = []\n",
+    "    df_purpose_trip = process_data_frame(expanded_ct, 'Trip_purpose', \"Labeled by user\\n (Confirmed trips)\")\n",
+    "    df_purpose_trip_eb = process_data_frame(data_eb, 'Trip_purpose', f\"Labeled `{mode_of_interest}` by user\\n (Replaced trips)\")\n",
```

Comment:
For the purpose chart, I think the purpose is for trips actually taken by e-bike so it's not really a replaced trip. For mode they are replaced trips because we're showing "40% of e-bike trips were replacing a car trip" but this chart is a little simpler in that it just shows "10% of e-bike trips were taken to go shopping" . I would omit the "(Replaced trips)" to avoid confusion